### ISSUE TITLE
Immediate VM crash causes hang on connecting

### ISSUE THREAD

**Issue by angrycub** [lab47/isle#2]
Hung on the Connecting step.  Eventually returned the following panic.
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x60 pc=0x100e3fc68]

goroutine 36 [running]:
github.com/lab47/yalr4m/pkg/crypto/ssh.NewClient.func1(0x140003ab090)
	/Users/evan/git/yalr4m/pkg/crypto/ssh/client.go:62 +0x28
created by github.com/lab47/yalr4m/pkg/crypto/ssh.NewClient
	/Users/evan/git/yalr4m/pkg/crypto/ssh/client.go:61 +0xf0
```

Tried to get some verbose logging for you
```
linux --start -V
2022-03-29T14:19:29.288-0400 [INFO]  linux: creating virtual machine: cores=10 memory=""
2022-03-29T14:19:29.381-0400 [INFO]  linux: observed vm state: state=4
2022-03-29T14:19:29.381-0400 [INFO]  linux: observed vm state: state=1
2022-03-29T14:19:29.381-0400 [INFO]  linux: vm detected as running, forwarding IO
2022-03-29T14:19:29.832-0400 [INFO]  linux: observed vm state: state=3
```
Seems to hang there.

In another session, tried to start again with verbose 

```
2022-03-29T14:21:18.443-0400 [INFO]  linux: connecting to unix socket
2022-03-29T14:21:18.443-0400 [INFO]  linux: negotiating ssh
```
get these two log lines repeating

In the first session I see this log line corresponding  to each attempt:
```
2022-03-29T14:21:18.444-0400 [ERROR] linux: attempted connection to session before started
```

Running linux-v0.2.3-arm64 on macOS 12.2.1 M1

**Comment by angrycub** [comment]
Adding info from conversation.  When I try to start os-v0.2.3-arm64.tar.gz, it looks like the VM is falling over immediately with a Virtualization.framework crash with the reason `FIXME: "Unhandled system register S3_0_C0_C6_2"`:

```
-------------------------------------
Translated Report (Full Report Below)
-------------------------------------
Process:               com.apple.Virtualization.VirtualMachine [63356]
Path:                  /System/Library/Frameworks/Virtualization.framework/Versions/A/XPCServices/com.apple.Virtualization.VirtualMachine.xpc/Contents/MacOS/com.apple.Virtualization.VirtualMachine
Identifier:            com.apple.Virtualization.VirtualMachine
Version:               1.0 (59.6)
Build Info:            Virtualization-59006000000000~7
Code Type:             ARM-64 (Native)
Parent Process:        launchd [1]
Responsible:           linux [63351]
User ID:               503
Date/Time:             2022-03-29 14:47:32.9758 -0400
OS Version:            macOS 12.2.1 (21D62)
Report Version:        12
Time Awake Since Boot: 19000 seconds
Time Since Wake:       18264 seconds
System Integrity Protection: enabled
Crashed Thread:        17  com.apple.virtualization.thread.cpu-0
Exception Type:        EXC_BREAKPOINT (SIGTRAP)
Exception Codes:       0x0000000000000001, 0x000000010089a408
Exception Note:        EXC_CORPSE_NOTIFY
Termination Reason:    Namespace SIGNAL, Code 5 Trace/BPT trap: 5
Terminating Process:   exc handler [63356]
Application Specific Information:
Generic Platform
FIXME: "Unhandled system register S3_0_C0_C6_2"
```

**Comment by evanphx** [comment]
I think this might be the same issue as https://communities.vmware.com/t5/Fusion-for-Apple-Silicon-Tech/Fedora-35-kernel-5-16-14-will-not-boot/td-p/2899003/page/2. I've given @angrycub a 5.16.11 kernel to try and he'll report back if that kernel works. If so, I'll release a 0.2.4 with that kernel instead of 5.16.17.

**Comment by evanphx** [comment]
This was a kernel <=> Virtualization.framework issue. v0.2.4 was released with kernel 5.16.11 to fix the issue.

**Comment by ghost** [comment]
If you need a quick way to just build kernels to drop in for testing, I put together this docker file a while back: https://github.com/protosam/ezkernel

It's not modular yet, but it's small enough to be easy to modify and compile time doesn't take forever.
